app/controllers/crosswalk_issues_controller.rb

```diff
@@ -47,17 +47,20 @@ def find_matches
     @issue = CrosswalkIssue.find(params[:id])
     address_data_to_match = @issue.weam.address_values.join
     physical_address_data = @issue.weam.physical_address_values.join
-    query = 'similarity(institution, ?) > 0.5 OR similarity((city||state||zip||addr), ?) > 0.3' \
-            'OR similarity((city||state||zip||addr), ?) > 0.3'
     escaped_institution = ApplicationRecord.connection.quote(@issue.weam.institution)
-    sanitize_order = IpedsHd.sanitize_sql_for_order("similarity(institution, #{escaped_institution}) DESC,
-                                         similarity((city||state||zip||addr), '#{address_data_to_match}') DESC,
-                                         similarity((city||state||zip||addr), '#{physical_address_data}') DESC")
 
-    @ipeds_hd_arr = IpedsHd.where(query, "%#{@issue.weam.institution}%",
-                                  "%#{address_data_to_match}%",
-                                  "%#{physical_address_data}%")
-                           .order(sanitize_order)
+    str = <<-SQL
+        SELECT id, ipeds_hds.cross, institution, addr, state, city, zip, ope, GREATEST(similarity(institution, #{escaped_institution}),
+                      similarity((city||state||zip||addr), '#{address_data_to_match}')
+                  , similarity((city||state||zip||addr), '#{physical_address_data}')) AS match_score
+        FROM ipeds_hds
+        WHERE (similarity(institution,  #{escaped_institution}) > 0.5
+          OR similarity((city||state||zip||addr), '#{address_data_to_match}') > 0.3
+          OR similarity((city||state||zip||addr), '#{physical_address_data}') > 0.3)
+        ORDER BY match_score DESC
+    SQL
+    sql = Institution.send(:sanitize_sql, str)
```

Comment:
`santize_sql` is already available on a Model, so send isn't necessary.  I think it makes more sense to call it against the model you're working with, so `IpedsHd.sanitize_sql(str)` makes a bit more sense to me.
I was reading up on the docs for `sanitize_sql` https://api.rubyonrails.org/v5.2.3/classes/ActiveRecord/Sanitization/ClassMethods.html and I don't think it's really necessary here.  I think it makes more sense to use the `quote` method like `escaped_institution` uses above.  Then you could even just call the `str` variable `sql` and leave out this sanitize line.

Reply:
I believe we need the `sanitize_sql ` call. For one thing, CodeClimate finds a possible sql injection issue without it. Which means we wouldn't be able to merge in the changes. From what I gathered, the fix needed for this issue is to sanitize the string. 

The use of `Institution` was a typo on my part. You are right, we should definitely use the model we are working in, though I don't think it really matters as long as it's a child of `ActiveRecord`.

Reply:
That's sort of annoying of code climate, because these values are from a model and not from user input.  But I'm new to VSP, so I didn't know it was going to flag that.

Reply:
I think `sanitize_sql` is intended to work only on a `WHERE` clause, so I'd personally be cautious about applying it to the whole thing.

Reply:
I'll point this out to my team for the refactor story we have for next sprint. Maybe we can find an alternative approach from `sanitize_sql`.